samples_project/Assets/SampleViewer/Samples/Viewshed/Scripts/ViewshedObserver.cs

```diff
@@ -0,0 +1,100 @@
+    [SerializeField] private Slider rotationSlider;
+    [SerializeField] private Slider opacitySlider;
+
+    private bool isHDRP;
+
+    private RenderTexture viewshedObserverDepthTexture;
+
+    void Start()
+    {
+        isHDRP = GraphicsSettings.currentRenderPipeline.GetType().ToString().Contains("HighDefinition");
+        SetupObserverCamera();
+        InitializeSliderValues();
+    }
+
+    void Update()
+    {
+        Shader.SetGlobalVector("_ViewshedObserverPosition", transform.position);
+        Shader.SetGlobalMatrix("_ViewshedObserverViewProjMatrix", viewshedObserverCamera.projectionMatrix * viewshedObserverCamera.worldToCameraMatrix);
+        Shader.SetGlobalTexture("_ViewshedObserverDepthTexture", viewshedObserverDepthTexture);
+        Shader.SetGlobalFloat("_ViewshedDepthThreshold", 0.00001f); 
+
+        // Generate the scaledScreenParams for HDRP (set by Unity by default in URP)
+        if(isHDRP)
+        {
+            Rect pixelRect = viewshedObserverCamera.pixelRect;
+            Shader.SetGlobalVector("_ViewshedObserverScreenParams", new Vector4(pixelRect.width, pixelRect.height, 1.0f + 1.0f / pixelRect.width, 1.0f + 1.0f / pixelRect.height));            
+        }
+    }
+
+    // Initialize the UI sliders to be in sync with the observer
+    private void InitializeSliderValues()
+    {
+        cameraFOVSlider.value = viewshedObserverCamera.fieldOfView;
+        altitudeSlider.value = transform.position.y;
+        rotationSlider.value = transform.rotation.eulerAngles.y;
+        opacitySlider.value = 0.5f;
+
+        cameraFOVSlider.onValueChanged.AddListener(delegate {
+            viewshedObserverCamera.fieldOfView = cameraFOVSlider.value;
+        });
+        altitudeSlider.onValueChanged.AddListener(delegate {
+            transform.position = new Vector3(transform.position.x, altitudeSlider.value, transform.position.z);
+        });
+        rotationSlider.onValueChanged.AddListener(delegate {
+            transform.rotation = Quaternion.Euler(transform.rotation.eulerAngles.x, rotationSlider.value, transform.rotation.eulerAngles.z);
+        });
+        opacitySlider.onValueChanged.AddListener(delegate {
+            Shader.SetGlobalFloat("_ViewshedOpacity", opacitySlider.value);
+        });
+        Shader.SetGlobalFloat("_ViewshedOpacity", opacitySlider.value);
+    }
+
+    // Setup/initialization related to the observer's camera
+    private void SetupObserverCamera()
+    {
+        if(viewshedObserverCamera != null)
+        {
+            viewshedObserverCamera.depthTextureMode = DepthTextureMode.Depth;
+            CreateViewshedObserverDepthTexture();
+            viewshedObserverCamera.targetTexture = viewshedObserverDepthTexture;
+        }
+    }
+
+    // Create the observer camera's depth texture (needed by main camera shader to render viewshed effect)
+    private void CreateViewshedObserverDepthTexture()
+    {
+        int w = 2048;
+        int h = 2048;
```

Comment:
fixed size? shouldn't this at least have the aspect ratio of the observer's view?